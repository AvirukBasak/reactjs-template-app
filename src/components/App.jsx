import React from 'react';
import printHelloWorld from '@/scripts/script';
import styles from '@/styles/App.module.css';

export default function App({ page }) {
  printHelloWorld();
  return (
    <>
      <div className={styles.app}>
        <h1>Demo App</h1>
        <p>This is a Paragraph</p>
        <img src="/images/background.jpg" />
      </div>
    </>
  );
}
