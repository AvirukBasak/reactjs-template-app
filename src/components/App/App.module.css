.app {
    width: 100%;
    height: 100vh;
    padding: 20px;

    display: flex;
    flex-direction: column;
    justify-content: space-evenly;

    background-color: rgb(8, 60, 60);

    background-image:
        linear-gradient(to bottom,
            rgba(45, 3, 47, 0.834),
            rgba(11, 11, 60, 0.788)),
        url('@/images/background.jpg');

    background-repeat: no-repeat;
    background-size: cover;
}

.app h1 {
    @apply text-4xl text-white;
    color: rgb(169, 169, 197);
    padding: 20px 0;
}

.app p {
    color: white;
}

.app img {
    width: 50%;
}
