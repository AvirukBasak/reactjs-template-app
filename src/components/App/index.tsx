import React from 'react'
import { printHelloWorld } from '@/utils/script'

import background from '@/images/background.jpg'
import styles from './App.module.css'

const App = (): JSX.Element => {
  printHelloWorld()
  return (
    <div className={styles.app + ' font-bold'}>
      <h1>Demo App</h1>
      <p>This is a Paragraph</p>
      <img src={background} />
    </div>
  )
}

export default App
