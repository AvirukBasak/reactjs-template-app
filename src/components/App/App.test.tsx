import React from 'react'
import App from '.'
import { render } from '@testing-library/react'

describe('App', () => {
  it('renders without crashing', () => {
    const wrapper = render(<App />)
    expect(wrapper).toBeTruthy()
  })
})
