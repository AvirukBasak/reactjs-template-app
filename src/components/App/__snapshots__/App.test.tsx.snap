// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="undefined font-bold"
>
  <h1>
    Demo App
  </h1>
  <p>
    This is a Paragraph
  </p>
  <img
    src={{}}
  />
</div>
`;
